### ISSUE TITLE
Feature request : pre hook before change

### ISSUE THREAD

**Issue by wbx13** [saltstack/salt#34039]
Hi there,

I'm planning to use Saltstack in my environment but I feel like there is no option to hook something just before a change is made.

Some of my servers are embeded servers and have their root file system mounted as read-only.
Because of this, I have to remount read-write before running the states, then remount read-only after the states have been run. I don't like being remounting rw every time a state is applied and doesn't run any change.

I think it would be nice to have a way to run a cmd or state just before changing the configuration, only if a change is required.

Thanks.
Cyril

**Comment by tbaker57** [comment]
Hi,

This sounds exacly what the 'prereq' requisite is for

https://docs.saltstack.com/en/latest/ref/states/requisites.html#prereq

cheers,
Tim

**Comment by wbx13** [comment]
Oh yes you are right ! Sorry for this.
